addons/sale_loyalty/models/sale_order.py

```diff
@@ -986,3 +989,18 @@ def _try_apply_code(self, code):
                 return apply_result
             coupon = apply_result.get('coupon', self.env['loyalty.card'])
         return self._get_claimable_rewards(forced_coupons=coupon)
+
+    def _get_loyalty_points(self):
+        loyalty_points_changes = defaultdict(lambda: 0)
+        for coupon_point in self.coupon_point_ids:
+            if coupon_point.coupon_id.program_type != 'loyalty':
+                continue
+            current = defaultdict(lambda: 0)
+            current['name'] = coupon_point.coupon_id.point_name
+            current['won'] = coupon_point.points
+            loyalty_points_changes[coupon_point.coupon_id] = current
+        for line in self.order_line:
+            if not line.reward_id or not line.coupon_id:
+                continue
+            loyalty_points_changes[line.coupon_id]['spent'] = line.points_cost
+        return loyalty_points_changes
```

Comment:
`name` will be empty and throw an error if all you do is spend points.

Reply:
I tested just spending points but could not get it to throw an error.
Could you tell me how to produce this situation?

Reply:
Nevermind it won't throw an error since you use `.get` to get the value but it will be empty in this case.